readium/navigator/src/main/java/org/readium/r2/navigator/pdf/PdfNavigatorFragment.kt

```diff
@@ -168,23 +137,23 @@ class PdfNavigatorFragment private constructor(
                 old.locator.href == new.locator.href
             }
             .map { state ->
-                createPdfDocumentFragment(state.locator, state.appliedSettings)
+                createPdfDocumentFragment(state.locator, settings.value)
             }
             .stateIn(viewLifecycleOwner.lifecycleScope, started = SharingStarted.Eagerly, null)
 
         viewLifecycleOwner.lifecycleScope.launch {
             viewLifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
                 documentFragment
                     .filterNotNull()
-                    .onEach { fragment ->
-                        childFragmentManager.commit {
+                    .onEach { fragment: PdfDocumentFragment<S> ->
+                        childFragmentManager.commitNow {
                             replace(R.id.readium_pdf_container, fragment, "readium_pdf_fragment")
                         }
                     }
                     .launchIn(this)
 
-                viewModel.state
-                    .map { it.appliedSettings }
+                settings
+                    .map { it }
```

Comment:
```suggestion
```

Reply:
That was needed to use `distinctUntilChange` without error. It looks like `distinctUntilChange` has no effect on `StateFlow` though, so it doesn't make sense.

Reply:
Ha, `distinctUntilChange` is not needed on a `StateFlow`, it already behaves like this. See "Operator fusion" here: https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/-state-flow/